handlers/tasks.go

```diff
@@ -0,0 +1,318 @@
+	// Декодируем JSON из тела запроса
+	var task models.Task
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if task.Title == "" {
+		http.Error(w, `{"error": "Не указан заголовок"}`, http.StatusBadRequest)
+		return
+	}
+
+	// Если дата не указана или указана как "today", ставим сегодняшнюю
+
+	if task.Date == "" || task.Date == "today" {
+		task.Date = time.Now().Format("20060102") // Текущая дата без времени (год, месяц, день)
+	}
+	parsedDate, err := time.Parse("20060102", task.Date)
+	if err != nil {
+		http.Error(w, `{"error": "Дата указана в неверном формате"}`, http.StatusBadRequest)
+		return
+	}
+
+	if parsedDate.Before(time.Now()) {
+		// Если правило повторения указано, пересчитываем дату
+		if task.Repeat != "" {
+			nextDate, err := NextDate(time.Now(), task.Date, task.Repeat)
+			if err != nil {
+				http.Error(w, fmt.Sprintf(`{"error": "%v"}`, err), http.StatusBadRequest)
+				return
+			}
+			task.Date = nextDate // Обновляем дату с учетом правила повторения
+		} else {
+			// Если правила повторения нет, устанавливаем текущую дату
+			task.Date = time.Now().Format("20060102")
+		}
+	}
+
+	// Проверяем правило повторения, если оно указано
+	if task.Repeat != "" {
+		// Проверка на соответствие формату repeat, например: "d <целое число>", "w <целое число>", "m <целое число>"
+		repeatPattern := `^([dwmy])\s?\d*$`
+		matched, err := regexp.MatchString(repeatPattern, task.Repeat)
+		if err != nil || !matched {
+			http.Error(w, `{"error": "Некорректное правило повторения"}`, http.StatusBadRequest)
+			return
+		}
+	}
+
+	// Если дата задачи меньше текущей
+	today := time.Now().Truncate(24 * time.Hour)
+	if parsedDate.Before(today) {
+		if task.Repeat == "" {
+			// Если правило повторения не указано, устанавливаем дату на сегодня
+			parsedDate = today
+		} else {
+			// Иначе вычисляем следующую допустимую дату
+			nextDate, err := NextDate(today, task.Date, task.Repeat)
+			if err != nil {
+				http.Error(w, `{"error": "Ошибка вычисления следующей даты"}`, http.StatusBadRequest)
+				return
+			}
+			parsedNextDate, err := time.Parse("20060102", nextDate)
+			if err != nil {
+				http.Error(w, `{"error": "Ошибка преобразования даты"}`, http.StatusBadRequest)
+				return
+			}
+			parsedDate = parsedNextDate
+		}
+	}
+
+	// Преобразуем дату обратно в строку
+	task.Date = parsedDate.Format("20060102")
+
+	// Проверяем правило повторения на корректность
+	validRepeat := false
+	validFormats := []string{"d", "w", "m", "y"}
+
+	for _, validFormat := range validFormats {
+		if strings.HasPrefix(task.Repeat, validFormat) {
+			validRepeat = true
+			break
+		}
+	}
+
+	if !validRepeat && task.Repeat != "" {
+		http.Error(w, `{"error": "Некорректное правило повторения"}`, http.StatusBadRequest)
+		return
+	}
+
+	// Добавляем отладочные сообщения
+	fmt.Println("Добавление задачи. Дата задачи:", task.Date)
```

Comment:
Можно всю структуру целиком выводить
```suggestion
	fmt.Printf("Добавление задачи. Дата задачи: %+v\n", task)
```